### PR TITLE
Switch from "invoice" to "order details" in our emails.

### LINKED ISSUE #26692: Remove "invoice" from the customer invoice email
The "customer invoice email" doesn't really send an invoice.

An invoice is a fiscal/legal document that you issue either before payment (if you sell on credit) or after payment (if you sell on pre-payment) and this email is not such a document.

In some countries, an invoice must be issued (by law) by a certified software (local or on the cloud), and calling this email "customer invoice email" can be misleading and, can be even become a legal issue (calling invoice to something is not).

I know that in some countries anything with a value on it could be considered an invoice, but I don't think this email qualifies as such a document. At most, it's a "payment request email" or an "order details email".

My proposal:
- Change the email name from "Customer invoice / Order details" to "Order details"
- Remove all mentions to "invoice" on this email (and WooCoommerce altogether)

### PR DESCRIPTION
**Pull request by barryhughes**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

We refer to some of our order emails as 'invoices', which turns out to be problematic for some merchants/in some jurisdictions. This change drops the use of 'invoice', preferring 'order details' instead.

Some existing uses of 'invoice' remain in place (for instance, in relation to class and template names) to preserve backward compatibility and minimize disruption.

Closes #26692.

### How to test the changes in this Pull Request:

> :bulb: When testing this, you will benefit from the ability to see the emails dispatched by WooCommerce. So long as your site is configured correctly, 'real emails' are fine for this—otherwise you may benefit from a tool such as [Email Log](https://wordpress.org/plugins/email-log/).

Open an existing order for editing, and look at the order actions meta box (typically found in the top right of the screen). You should see an entry reading, *"Send order details to customer."* Previously, this read, *"Email invoice / order details to customer."*

<div align="center">Order actions meta box, showing the 'send order details to customer' option.</div>

Select the above option and click on *Update.* The email should be sent to the customer:

- The email subject should read `Details for order #1234 on My Site` (of course, the order number and site name will be different). Previously, it was either `Invoice for order #1234 on My Site` or `Your latest My Site invoice` (depending on order payment status).
- The email header should simply be `Details for order #1234`. Previously, it was either `Invoice for order #1234` or `Your invoice for order #1234`.

<div align="center">Part of the order details email.</div>

Now visit **WooCommerce ▸ Settings ▸ Emails**. You should see an entry for the `Order details` email (previously was referred to as `Customer invoice / Order details`):

<div align="center">Section of the order admin user interface, showing the newly renamed 'Order details' email.</div>

Finally, consider clicking through via the above admin screen and modifying the settings for this email. Your changes should be reflected just as they would have been before this PR was written.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

**Comment by barryhughes** [comment]
As was previously true, the email subject and header text can anyway be defined and overridden by the merchant (via the email settings screen). This change is really intended just to tweak the defaults, since use of 'invoice' is problematic in some contexts and geographies.

**Comment by webdados** [comment]
"Order details" on the order actions menu and as email name is perfect.

I would set the default email subject to "Details for order #1245 on My Site" and "Details for order #1245" as header.

File: plugins/woocommerce/includes/emails/class-wc-email-customer-invoice.php [hunk 1]
```diff
@@ -52,11 +54,7 @@ public function __construct() {
 		 * @return string
 		 */
 		public function get_default_subject( $paid = false ) {
-			if ( $paid ) {
-				return __( 'Invoice for order #{order_number} on {site_title}', 'woocommerce' );
-			} else {
-				return __( 'Your latest {site_title} invoice', 'woocommerce' );
-			}
+			return __( 'Order #{order_number} from {site_title}', 'woocommerce' );
```
Comment (barryhughes) [inline plugins/woocommerce/includes/emails/class-wc-email-customer-invoice.php]:
```suggestion
			return __( 'Details for order #{order_number} on {site_title}', 'woocommerce' );
```

Via [feedback](https://github.com/woocommerce/woocommerce/pull/45395#issuecomment-1983913935) from @webdados.

File: plugins/woocommerce/includes/emails/class-wc-email-customer-invoice.php [hunk 2]
```diff
@@ -67,11 +65,7 @@ public function get_default_subject( $paid = false ) {
 		 * @return string
 		 */
 		public function get_default_heading( $paid = false ) {
-			if ( $paid ) {
-				return __( 'Invoice for order #{order_number}', 'woocommerce' );
-			} else {
-				return __( 'Your invoice for order #{order_number}', 'woocommerce' );
-			}
+			return __( 'Order #{order_number}', 'woocommerce' );
```
Comment (barryhughes) [inline plugins/woocommerce/includes/emails/class-wc-email-customer-invoice.php]:
```suggestion
			return __( 'Details for order #{order_number}', 'woocommerce' );
```

Via [feedback](https://github.com/woocommerce/woocommerce/pull/45395#issuecomment-1983913935) from @webdados.

File: plugins/woocommerce/tests/e2e-pw/tests/merchant/order-emails.spec.js [hunk 3]
```diff
@@ -237,7 +237,7 @@ test.describe( 'Merchant > Order Action emails received', () => {
 				.getByRole( 'row' )
 				.filter( { hasText: customerBilling.email } )
 				.filter( {
-					hasText: `Invoice for order #${ orderId } on ${ storeName }`,
+					hasText: `Order #${ orderId } from ${ storeName }`,
```
Comment (barryhughes) [inline plugins/woocommerce/tests/e2e-pw/tests/merchant/order-emails.spec.js]:
```suggestion
					hasText: `Details for order #${ orderId } on ${ storeName }`,
```

To match earlier change in wording.

**Comment by barryhughes** [comment]
> I would set the default email subject to "Details for order https://github.com/woocommerce/woocommerce/pull/1245 on My Site" and "Details for order https://github.com/woocommerce/woocommerce/pull/1245" as header.

Thanks, @webdados: updated.

**Review by jorgeatorres — APPROVED** [review]
LGTM. Thank you @barryhughes! Tests well.

Adding that I pushed a tiny commit to fix a typo in a docblock (f608530dd47fc495f322d505a1a6d6fac0092eef).

**Comment by webdados** [comment]
Thanks for this!

**Pull request merged**